src/app/dynamic/enterprise/quotas/component.spec.ts

```diff
@@ -0,0 +1,205 @@
+  });
+
+  it('should not display table when quotas are loaded empty', async () => {
+    component.isLoading = false;
+    component.quotas = [];
+    fixture.detectChanges();
+
+    const element = fixture.debugElement.query(By.css('table'));
+
+    expect(element.nativeElement.hidden).toEqual(true);
+  });
+
+  it('should not display loading spinner when quotas are loading', () => {
+    component.isLoading = true;
+    fixture.detectChanges();
+
+    const element = fixture.debugElement.query(By.css('#quotas-loading-spinner'));
+
+    expect(element).toBeTruthy();
+  });
+
+  it('should hide loading spinner when quotas are loaded', () => {
+    component.isLoading = false;
+    fixture.detectChanges();
+
+    const element = fixture.debugElement.query(By.css('#quotas-loading-spinner > mat-spinner'));
+
+    expect(element).toBeFalsy();
+  });
+
+  it('should display correct message when loaded quotes are empty', () => {
+    component.isLoading = false;
+    component.quotas = [];
+    fixture.detectChanges();
+
+    const element = fixture.debugElement.query(By.css('#quotes-not-found'));
+
+    expect(element.nativeElement.textContent.trim()).toEqual('No Quotas Found');
+  });
+
+  it('should display paginator when received quotes are more than the user defined page size', () => {
+    const mockLength = 5;
+    component.quotas.length = mockLength - 1;
+    component.paginator.pageSize = mockLength;
+    fixture.detectChanges();
+
+    const element = fixture.debugElement.query(By.css('#quotas-paginator'));
+
+    expect(element.nativeElement.hidden).toEqual(true);
+  });
+
+  it('should hide paginator when received quotes are less than the user defined page size', () => {
+    const mockLength = 5;
+    component.quotas.length = mockLength;
+    component.paginator.pageSize = mockLength + 1;
+    fixture.detectChanges();
+
+    const element = fixture.debugElement.query(By.css('#quotas-paginator'));
+
+    expect(element.nativeElement.hidden).toEqual(true);
+  });
+
+  it('should set quotas when ngOnInit is called', async () => {
+    const quotas = await getQuotas();
+
+    component.ngOnInit();
+
+    expect(component.quotas).toEqual(quotas);
+  });
+
+  it('should set user when ngOnInit is called', async () => {
+    component.user = null;
+    const user = await getUser();
+
+    component.ngOnInit();
+
+    expect(component.user).toStrictEqual(user);
+  });
+
+  it('should set paginator.pageSize when ngOnInit is called', async () => {
+    const userSettings = await getUserSetting();
+
+    component.ngOnInit();
+
+    expect(component.paginator.pageSize).toEqual(userSettings.itemsPerPage);
+  });
+
+  it('should call setDataSource() when ngOnChanges() is called', async () => {
+    const spy = jest.spyOn(component, 'ngOnChanges');
+
+    component.ngOnChanges();
+
+    expect(spy).toHaveBeenCalled();
+  });
+
+  it('should set dataSource.data when setDataSource() called', async () => {
```

Comment:
Better to check if the value is rendered in table or not.

Reply:
adding additional unit tests belonging to table rendering